src/gda/bnxt/queue_pair_bnxt.cpp

```diff
@@ -391,6 +398,10 @@ __device__ uint64_t QueuePair::post_wqe_amo(int pe, int32_t length, uintptr_t *r
     release_lock(&sq.lock);
   }
 
+  if (fetching) {
+    return fetching_atomic[atomic_idx];
```

Comment:
https://github.com/ROCm/rocSHMEM/pull/253/files#diff-2548ac2b0900f38b49bf1847b485ba1f7d535eef55217960acb02e291c0a1a1dR394

could we move that threadfence and quiet in the fetching if here?